bin/alpen-bridge/src/rpc_server.rs

```diff
@@ -231,57 +345,40 @@ impl StrataBridgeMonitoringApiServer for BridgeRpc {
 
     async fn get_bridge_duties(&self) -> RpcResult<Vec<RpcBridgeDutyStatus>> {
         // we don't care about the hot state here, we only care about the database
-        let all_entries = query!(r#"SELECT * FROM contracts"#)
-            .fetch_all(self.db.pool())
-            .await
-            .map_err(|_| {
-                ErrorObjectOwned::owned::<_>(
-                    -666,
-                    "Database error. Config dumped",
-                    Some(self.db.config()),
-                )
-            })?;
+        let all_entries = self.cached_contracts.read().await.clone();
 
         let mut duties = Vec::new();
         for entry in all_entries {
-            let state = bincode::deserialize::<ContractState>(&entry.state).map_err(|_| {
-                ErrorObjectOwned::owned::<_>(
-                    -666,
-                    "Database error. Config dumped",
-                    Some(self.db.config()),
-                )
-            })?;
-
-            match state {
+            match &entry.0.state {
                 ContractState::Requested {
                     deposit_request_txid,
                     ..
                 } => {
                     duties.push(RpcBridgeDutyStatus::Deposit {
-                        deposit_request_txid,
+                        deposit_request_txid: *deposit_request_txid,
```

Comment:
again this is not a property of state, this should be pulled from config.

Reply:
Fixed in f6b267f964ad9ecb4ec1dde7f23d09376c08c013